components/docker-auth/pkg/extension/acl.go

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright (c) 2019 WSO2 Inc. (http:www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http:www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package extension
+
+import (
+	"database/sql"
+	"encoding/json"
+	"errors"
+	"log"
+	"net"
+	"strings"
+
+	_ "github.com/go-sql-driver/mysql"
+)
+
+type Labelstest map[string][]string
+
+type AuthRequestInfo struct {
+	Account string
+	Type    string
+	Name    string
+	Service string
+	IP      net.IP
+	Actions []string
+	Labels  Labelstest
+}
+
+func ValidateAccess(db *sql.DB, accessToken string) (bool, error) {
+	var authReqInfo AuthRequestInfo
+	err := json.Unmarshal([]byte(accessToken), &authReqInfo)
+	if err != nil {
+		log.Println("Unable to unmarshal the json :", err)
+		return false, err
+	}
+	log.Println("Required actions for the username are :", authReqInfo.Actions)
+	isPullOnly := true
+	if len(authReqInfo.Actions) == pushActionCount {
+		isPullOnly = false
+	}
+	organization, image := getOrganizationAndImage(authReqInfo.Name)
+	log.Println("Image name is declared as :", image)
+	if isPullOnly {
+		log.Println("Received a pulling task")
+		return isAuthorizedToPull(db, authReqInfo.Account, organization, image)
+	} else {
+		log.Println("Received a pushing task")
+		return isAuthorizedToPush(db, authReqInfo.Account, organization)
+	}
+}
+
+func getOrganizationAndImage(imageFullName string) (string, string) {
+	tokens := strings.Split(imageFullName, "/")
```

Comment:
Shall we do a length validation after the split function. If the array size is less than 2 it might throw an index out of range error.

Reply:
Fixed